### ISSUE TITLE
New dev container is created every time the folder is openned

### ISSUE THREAD

**Issue by wallyhall** [microsoft/vscode-remote-release#9001]
Type: <b>Bug</b>

Open any folder in VSCode with a .devcontainer definition.
When prompted, "reopen in dev container" (or use the "Recent" list direct to dev container, etc).
VSCode creates a whole new dev container, and then complains that the workspace doesn't exist.

Have tried completely reinstalling VSCode, the plugin, and Docker Desktop.

Extension version: 0.309.0
VS Code version: Code 1.82.2 (abd2f3db4bdb28f9e95536dfa84d8479f1eb312d, 2023-09-14T05:55:25.390Z)
OS version: Windows_NT x64 10.0.22621
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz (12 x 2592)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.75GB (7.91GB free)|
|Process Argv|C:\\Development\\pegasus-state-snapshot-service --crash-reporter-id b30ba0a5-c8bb-46c0-b1c6-a55ca7aab9e7|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
vsclangdc:30486549
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
vsccc:30803844
3biah626:30602489
89544117:30613380
2i9eh265:30646982
showlangstatbar:30737416
962ge761:30835153
a2ce3375:30757347
pythonfmttext:30731395
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
pythonnosmt12:30797651
pythonidxpt:30805730
pythonnoceb:30805159
copilotsettingt:30839829
dsvsc013:30795093
dsvsc014:30804076
diffeditorv2:30821572
dsvsc015:30829745

```

</details>

**Comment by chrmarti** [comment]
- What is the absolute local path of the devcontainer.json you are using?
- Could you post the out from `docker inspect <container-id>` for one the created dev containers?

**Comment by wallyhall** [comment]
I'm sorry, in trying to get my local dev restored - I nuked VSCode's %APPDATA%, and that instantly fixed the fault.  I didn't back it up before doing so either :(

Whatever it was, I suspect there was a malformed or invalid value somewhere in that.  Once I re-setup VSCode (I just logged back into my MS account and let it resync everything "as it had been", the issue has been resolved.

I'm happy to close this issue - unless you wish to try and investigate it any further?
